### PR TITLE
getPostType(s) filter and fields

### PR DESCRIPTION
**Pull request by NathanKleekamp**
I added the getPostType fields param and getPostTypes filter and fields params, with docs. Let me know if anything needs to be changed. Thanks!

### CODE REVIEW

File: lib/wordpress.js [hunk 1]
```diff
@@ -187,9 +187,21 @@ extend( Client.prototype, {
 		this.authenticatedCall( "wp.deletePost", id, fn );
 	},
 
-	getPostType: function( name, fn ) {
-		// TODO: fields
-		this.authenticatedCall( "wp.getPostType", name, function( error, postType ) {
+	getPostType: function( name, fields, fn ) {
+		if ( typeof name !== "string" ) {
+			throw new TypeError("Expected first argument to be a post type (string). Recieved "+ typeof name +" instead");
+		}
```
Comment (scottgonzalez) [inline lib/wordpress.js]: We don't do checks like this anywhere else. I'd prefer consistency.

File: lib/wordpress.js [hunk 2]
```diff
@@ -198,10 +210,28 @@ extend( Client.prototype, {
...
+			fields = null;
+			filter = {};
+		}
+
+		if ( typeof fields === "function" ) {
+			fn = fields;
+			fields = null;
+		}
+
+		if ( Object.prototype.toString.call( filter ) === "[object Array]" ) {
```
Comment (scottgonzalez) [inline lib/wordpress.js]: Use `Array.isArray()`.

File: lib/wordpress.js (hunk 1 above)
Comment (NathanKleekamp) [inline lib/wordpress.js]: Does that mean you'd like me to add the error checking in the other places or remove it from here? Without it, you get a somewhat cryptic `TypeError: fn is not a function`

Comment (scottgonzalez) [inline lib/wordpress.js]: Let's remove it from here, but feel free to send another PR that adds it everywhere.

**Comment by NathanKleekamp** [comment]
OK, @scottgonzalez , I believe I'm made all the requested changes. Let me know if there's anything else. Thanks! -Nathan

**Pull request closed without merging**